src/test/java/com/softserveinc/dokazovi/service/impl/LogServiceImplTest.java

```diff
@@ -89,4 +96,29 @@ void findAllPostLogsWithTimestamps() {
                 any(Timestamp.class), any(Timestamp.class));
         verify(logMapper, times(2)).toPostLogDTO(any(LogEntity.class));
     }
+
```

Comment:
Just another thing for a follow-up PR, https://docs.spring.io/spring-framework/reference/testing/testcontext-framework/application-events.html

It lets you verify that an event has been fired and will be processed by spring.

And then onApplicationEvent method can be tested in a separate test class.

Reply:
Also, just a codestyle nit, also not critical - it is a better idea to have some spacing between contextually different parts of code. E.g.:

```java
        UserEntity expected = UserEntity.builder()
                .id(1)
                .firstName("First Name")
                .lastName("Last Name")
                .email("admin@mail.com")
                .password("$2y$10$GtQSp.P.EyAtCgUD2zWLW.01OBz409TGPl/Jo3U30Tig3YbbpIFv2")
                .build();

        LogEntity log = LogEntity.builder()
                .title("Test")
                .changes("Матеріал видалено")
                .idOfChangedPost(1)
                .nameOfChanger(expected.getLastName() + " " + expected.getFirstName())
                .build();

        when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(expected));
        when(logRepository.save(any(LogEntity.class))).thenReturn(log);

        logService.makeEntryInLogs(log.getTitle(), UserPrincipal.create(expected), log.getChanges(),
                log.getIdOfChangedPost());

        verify(logRepository, times(1))
                .save(any(LogEntity.class));

        assertEquals(1, log.getIdOfChangedPost());
        assertEquals(expected.getLastName() + " " + expected.getFirstName(), log.getNameOfChanger());
```

Also grouped variables in this example, as well just for sake of readability.